sw/device/tests/keymgr_sideload_aes_test.c

```diff
@@ -153,6 +159,33 @@ void aes_test(void) {
   CHECK_DIF_OK(dif_aes_read_output(&aes, &out_data_plain));
   CHECK_ARRAYS_NE(out_data_plain.data, kPlainText, ARRAYSIZE(kPlainText));
   CHECK_DIF_OK(dif_aes_end(&aes));
+
+  // Generate the same key again and check that encryption result is identical
+  // as the first.
+  keymgr_testutils_generate_versioned_key(&keymgr, sideload_params);
+  LOG_INFO("Keymgr generated HW output for Aes at OwnerIntKey State");
+
+  transaction.operation = kDifAesOperationEncrypt;
+  LOG_INFO(
+      "Encrypting the plaintext again with the same 256-bit AES sideload key.");
+  CHECK_DIF_OK(dif_aes_start(&aes, &transaction, NULL, NULL));
+  memcpy(in_data_plain.data, kPlainText, sizeof(kPlainText));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusInputReady, true, TIMEOUT);
+  CHECK_DIF_OK(dif_aes_load_data(&aes, in_data_plain));
+
+  // Trigger the AES encryption and wait for it to complete.
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerStart));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusOutputValid, true, TIMEOUT);
+
+  //  Verify that the ciphertext is identical to the first one generated.
+  dif_aes_data_t out_data_second;
+  CHECK_DIF_OK(dif_aes_read_output(&aes, &out_data_second));
+  CHECK_ARRAYS_EQ(out_data.data, out_data_second.data,
+                  ARRAYSIZE(out_data.data));
+
+  // Finish the ECB encryption transaction.
+  LOG_INFO("Finished encrypting again with the same key");
+  CHECK_DIF_OK(dif_aes_end(&aes));
```

Comment:
looks like this encrypt operation is replicated in 3 places. can we make it a function? perhaps `out_data` can be the function output and we can check it in `aes_test`

Reply:
yes, that makes sense. thanks, will take care of it.

Reply:
@weicaiyang I moved the encrypt/decrypt operations into one function and call it as you suggest. Let me know if it looks good now.